src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala

```diff
@@ -561,10 +561,38 @@ object Initializer extends SparkSessionWrapper {
   // Init the SparkSessionWrapper with envVars
   envInit()
 
-  def getOrgId: String = {
+  private def getApiURL(apiUrlInput: Option[String]): String = {
+    apiUrlInput match {
+      case Some(apiURL) => apiURL
+      case None => dbutils.notebook.getContext.apiUrl match {
+        case Some(apiURL) => apiURL
+        case None => throw new BadConfigException("API URL cannot be determined")
+      }
+    }
+  }
+
+  /**
+   * Returns the local workspace orgID
+   * @return
+   */
+  def getOrgId: String = getOrgId(None)
+
+  /**
+   * Returns the local workspace orgID, or the orgID contained in the input apiURL
+   * When getOrgId is called from a Future the apiURL MUST be passed to prevent failure to get orgID
+   * since dbutils.notebook.getContext.apiUrl.get returns None and None.get fails
+   * @param apiUrl apiURL of the workspace to get the orgId
+   * @return
+   */
+  def getOrgId(apiUrlInput: Option[String]): String = {
```

Comment:
We should also call this one from `Initializer.initConfigState`

Reply:
also from `Helpers.getWorkspaceByDatabase`

Reply:
nice catch, thank you!